swift/swift.py

```diff
@@ -106,29 +95,29 @@ def createApp(self, name: str, info: dict):
               path: A path desired to be added for importing app.
               module: A name of app module.
               class: A name of app class.
-              show: Whether its frames is shown at the beginning.
+              show: Whether its frames are shown at the beginning.
                 True if you want to show the frames, otherwise False.
               pos: A string that indicates the position of the frames.
                 It should be one of "left", "right", "top", or "bottom"; and is case-sensitive.
                 Otherwise, it will be regarded as default (AllDockWidgetAreas).
-              bus: A list of buses which app subscribes to.
-              args: Additional arguments for custom.
+              bus: A list of buses which the app subscribes to.
+              args: Additional arguments for app class constructor.
                 If there is no additional arguments, set None.
         """
         # import the app module
-        with _add_to_path(os.path.dirname(info.get("path", ""))):
+        with _add_to_path(os.path.dirname(info.get("path", "."))):
             module = importlib.import_module(info["module"])
         # create an app
         cls = getattr(module, info["class"])
         if "args" in info:
-            app = cls(name, self, **info["args"])
+            app = cls(name, parent=self, **info["args"])
         else:
             app = cls(name, self)

```

Comment:
Now I realized that it should be `parent=self` here as well.
Suppose the constructor of `cls` looks like
```python
def __init__(self, name, arg1=None, arg2=None, parent=None):
```
and `"args"` is not given in order to use the default values for `arg1` and `arg2`.

Reply:
I was talking about the line 115.
The line 113 is not a problem since the keyword argument order is not important.
However, as in line 115, if the argument is given as a positional argument, the `self` in the second position could be misgiven as another argument (`arg1`, not `parent` in my example above).